db/migrations/20210129201213_single_name_field.sql

```diff
@@ -1,15 +1,29 @@
 -- migrate:up
 -- Add new single name field
-ALTER TABLE users ADD COLUMN name VARCHAR;
-COMMENT ON COLUMN users.name IS 'Name of user, may or may not be full name.';
+ALTER TABLE users ADD COLUMN full_name VARCHAR;
+COMMENT ON COLUMN users.full_name IS 'Name of user, may or may not be full name.';
 
 -- Migrate names over
-UPDATE users SET name = users.first_name || ' ' || users.last_name;
+UPDATE users SET full_name = COALESCE(users.first_name, '') || ' ' || COALESCE(users.last_name, '');
 
-ALTER TABLE users ALTER COLUMN name SET NOT NULL;
+ALTER TABLE users ALTER COLUMN full_name SET NOT NULL;
 
 -- Drop old name fields
 ALTER TABLE users DROP COLUMN first_name CASCADE, DROP COLUMN last_name CASCADE;
 
+-- Display name is either full name or preferred name if set
+ALTER TABLE users
+  ADD COLUMN display_name VARCHAR
+    GENERATED ALWAYS AS (COALESCE(preferred_name, full_name)) STORED;
```

Comment:
Why don't we just let users set their name to whatever they prefer to be called? That way we don't have to have additional fields to discriminate their 'actual' name. Am I missing something here?